darts/timeseries.py

```diff
@@ -911,6 +935,160 @@ def univariate_values(self, copy=True, sample=0) -> np.ndarray:
+        component_axis_equal = len(set([ts.width for ts in timeserie_sequence])) == 1
+        sample_axis_equal = len(set([ts.n_samples for ts in timeserie_sequence])) == 1
+
+        if ((axis == 'time' and not (component_axis_equal and sample_axis_equal)) or
+            (axis == 'component' and not ((time_axis_equal or ignore_time_axes) and sample_axis_equal)) or
+            (axis == 'sample' and not ((time_axis_equal or ignore_time_axes) and component_axis_equal))):
+
+            raise_log(AttributeError('Remaining (non-concatenating) axes need to be equal.'))
+
+        da_sequence = [ts.data_array() for ts in timeserie_sequence]
+
+        if axis == DIMS[0]: # time
+            if time_axis_equal:
+                if ignore_time_axes:
+                    da_concat=da_sequence[0]
+                else:
+                    raise_log(AttributeError("All concatenated have the same time axis. "
+                                             "If this is not an error, use `ignore_time_axis=True`."))
+            else:
+                consecutive_time_axes = True
+                for i in range(1, len(timeserie_sequence)):
+                    if timeserie_sequence[i - 1].end_time() + pd.Timedelta(timeserie_sequence[0].freq) != \
+                            timeserie_sequence[i].start_time():
+                        consecutive_time_axes = False
+                        break
+
+                if consecutive_time_axes:
+                    da_concat = xr.concat(da_sequence, dim=axis)
+                else:
+                    raise_log(AttributeError("When concatenating over time axis, all series need to be subsequent, i.e."
+                                             " end_time of the previous one should be equal to start_time - 1 of the "
+                                             "following timeserie."
+                                         ))
+
+        else: # component or sample
+            if len(set([ts.shape[0] for ts in da_sequence])) != 1:
+                raise AttributeError("All concatenating time series need to have the same time axis or at least"
+                                     " be of the same length.")
+            if time_axis_equal:
+                da_concat = xr.concat(da_sequence, dim=axis)
+            else:
+                if ignore_time_axes:
+                    ts1_time_coord = da_sequence[0].coords[DIMS[0]]
+                    for i in range(1, len(da_sequence)):
+                        da_sequence[i] = da_sequence[i].assign_coords(time=ts1_time_coord)
+
+                    da_concat = xr.concat(da_sequence, dim=axis)
+                else:
+                    raise_log(AttributeError("All concatenating time series need to have the same time axis."
+                                             " You can override this error by setting `ignore_time_axes=True`."))
+
+        return TimeSeries.from_xarray(da_concat, fill_missing_dates=True, freq=timeserie_sequence[0].freq_str)
```

Comment:
I don't think you need to call the factory method here, probably the constructor is enough. Furthermore you for sure shouldn't set `fill_missing_dates=True` because this is expensive to compute and should not be needed at this stage. Also no need to set the freq if the time axis of the DataArray contains the correct freq already.